core/src/wallet/bitcoin/database/BitcoinLikeUTXODatabaseHelper.cpp

```diff
@@ -84,7 +97,44 @@ namespace ledger {
             return out.size();
         }
 
+        std::vector<BitcoinLikeBlockchainExplorerUtxo> BitcoinLikeUTXODatabaseHelper::queryAllUtxos(
+            soci::session &session, std::string const &accountUid, api::Currency const &currency)
+        {
+            soci::rowset<soci::row> rows = (
+                session.prepare <<
+                    "SELECT o.address, o.idx, o.transaction_hash, o.amount, o.script, o.block_height, i.sequence "
+                    "FROM bitcoin_outputs AS o "
+                    "LEFT OUTER JOIN bitcoin_inputs AS i ON i.previous_tx_uid = o.transaction_uid AND i.previous_output_idx = o.idx "
+                    "WHERE i.previous_tx_uid IS NULL AND o.account_uid = :uid "
+                    "ORDER BY o.block_height",
+                use(accountUid));
 
+            std::vector<BitcoinLikeBlockchainExplorerUtxo> filteredUtxo;
 
+            // create on the fly the row helper
+            for (soci_helper::Row row : rows) {
+                auto const outputAddress = row.getField(OUTPUT_ADDRESS);
+
+                if (outputAddress.hasValue()) {
+                    filteredUtxo.emplace_back(BitcoinLikeBlockchainExplorerUtxo{
+                       row.getField(OUTPUT_INDEX),
```

Comment:
I agree with you than the previous is more local (even if now the variables in the same TU) but I totally disagree than that save kind of errors. 
The legacy code is really hard to understand - what index 1 means ? Why the type is a unsigned integer ? In my version we relate on well-defined variables which contains a name so define what the column is. That implementation can be improved by creating a class than manage such a row and deal with indexes and so on.

Reply:
What is preventing you for making error in `OUTPUT_INDEX` definition? What is 1 in `OUTPUT_INDEX` definition? Why the type is unsigned integer? We move this problem to another place but we don't solve it. 
It may have sense if we can define this constants for all queries and then reuse them, but we can't as for each query the selected fields are different.

Reply:
Ok I got your point. I can move that `constexpr` variable into the function.